MacPass/MPPasswordInputController.m

```diff
@@ -81,24 +91,27 @@ - (void)viewDidLoad {
   [self.enablePasswordCheckBox bind:NSValueBinding toObject:self withKeyPath:NSStringFromSelector(@selector(enablePassword)) options:nil];
   [self.togglePasswordButton bind:NSEnabledBinding toObject:self withKeyPath:NSStringFromSelector(@selector(enablePassword)) options:nil];
   [self.passwordTextField bind:NSEnabledBinding toObject:self withKeyPath:NSStringFromSelector(@selector(enablePassword)) options:nil];
+  self.touchIdEnabled.hidden = true;
+  if (@available(macOS 10.13.4, *)) {
+    self.touchIdEnabled.hidden = false;
+    self.touchIdEnabled.state = [NSUserDefaults.standardUserDefaults integerForKey:kMPSettingsKeyEntryTouchIdEnabled];
+    [self _updateTouchIdTooltip];
+  }
   [self _reset];
 }
 
 - (NSResponder *)reconmendedFirstResponder {
   return self.passwordTextField;
 }
 
-- (void)requestPasswordWithMessage:(NSString *)message cancelLabel:(NSString *)cancelLabel completionHandler:(passwordInputCompletionBlock)completionHandler {
+- (void)requestPasswordWithMessage:(NSString *)message cancelLabel:(NSString *)cancelLabel completionHandler:(passwordInputCompletionBlock)completionHandler forFile:(NSURL*) fileURL{
```

Comment:
See comment on self.databaseFileURL an not having to store it since we can get to via the window controller.